orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java

```diff
@@ -194,7 +197,52 @@ private RollbackDetails findBestCandidate(
         previousImageRollbackSupport.getImageDetailsFromEntityTags(
             stageData.cloudProvider, stageData.credentials, region, serverGroupToRollBack.name);
 
-    return getBestCandidate(cluster, region, serverGroupToRollBack, candidates, imageDetails);
+    return getOriginalServerGroup(stage)
+        .map(
+            serverGroupToRestore ->
+                getRollbackDetails(serverGroupToRollBack.name, serverGroupToRestore, imageDetails))
+        .orElseGet(
+            () ->
+                getBestCandidate(cluster, region, serverGroupToRollBack, candidates, imageDetails));
+  }
+
+  private RollbackDetails getRollbackDetails(
+      String serverGroupToRollBack, String serverGroupToRestore, ImageDetails imageDetails) {
+
+    Map<String, String> context =
+        ImmutableMap.<String, String>builder()
+            .put("rollbackServerGroupName", serverGroupToRollBack)
+            .put("restoreServerGroupName", serverGroupToRestore)
+            .build();
+
+    return new RollbackDetails(
+        EXPLICIT, context, imageDetails.getImageName(), imageDetails.getBuildNumber());
+  }
+
+  /** Return the name of the original server group we should roll back to */
+  private Optional<String> getOriginalServerGroup(StageExecution stage) {
+    return Optional.ofNullable(
+            stage.findAncestor(isType(CreateServerGroupStage.PIPELINE_CONFIG_TYPE)))
```

Comment:
Reading the code, I discovered that the CloneServerGroupStage and the (deprecated) CopyLastAsgStage could also potentially be ancestors.  I'll need to revise this code to handle those cases.

Reply:
I'm going to use a different strategy for implementing this: I'm going to modify the MonitoredDeployStrategy to set the original server group as context on this stage, rather than calling findAncestor here. Updates coming, stay tuned...

Reply:
This is now done.